site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -112,4 +120,130 @@ public function showSummary() {
+        $submit_data['grader_id'] = isset($_POST['overwrite']) ? $this->core->getUser()->getId() : $gradeable->getGrader()->getId();
+        $submit_data['late_charged'] = intval($_POST['late']);
+        $submit_data['comment'] = $_POST['comment-general'];
+        $submit_data['active_version'] = $gradeable->getActiveVersion();
+        $submit_data['time'] = $now->format("Y-m-d H:i:s");
+        
+        $submit_data['components'] = array();
+        $comps = $gradeable->getComponents();
+        //update each gradeable component data
+        foreach($comps as $comp){
+            $gc_id = $comp->getId();
+            $submit_data['components'][$gc_id] = array();
+            $submit_data['components'][$gc_id]['grade'] = floatval($_POST["grade-{$comp->getOrder()}"]);
+            $submit_data['components'][$gc_id]['comment'] = isset($_POST["comment-{$comp->getOrder()}"]) ? $_POST["comment-{$comp->getOrder()}"] : '';
+        }
+
+        $this->core->getQueries()->submitTAGrade($submit_data);
+
+        $hwReport = new HWReport($this->core);
+        $hwReport->generateSingleReport($who_id, $gradeable_id);
+        $individual = intval($_POST['individual']);
+
+        $_SESSION['messages']['success'][] = "Successfully uploaded grade for {$who_id}";
+
+        if ($individual == 1) {
+            $this->core->redirect($this->core->buildUrl(array('component'=>'grading', 'page'=>'electronic', 'action'=>'summary','gradeable_id'=>$gradeable_id)));
+        }
+        else {
+            $this->core->redirect($this->core->buildUrl(array('component'=>'grading', 'page'=>'electronic', 'action'=>'grade', 'gradeable_id'=>$gradeable_id, 'individual'=>'0')));
+        }
+        
+    }
+
+    public function showGrading() {
+        $gradeable_id = $_REQUEST['gradeable_id'];
+        $gradeable = $this->core->getQueries()->getGradeable($gradeable_id);
+
+        if ($gradeable->isGradeByRegistration()) {
+            $sections = $this->core->getUser()->getGradingRegistrationSections();
+            $users_to_grade = $this->core->getQueries()->getUsersByRegistrationSections($sections);
+            $total = array_sum($this->core->getQueries()->getTotalUserCountByRegistrationSections($sections));
+            $graded = array_sum($this->core->getQueries()->getGradedUserCountByRegistrationSections($gradeable_id, $sections));
+        }
+        else {
+            $sections = $this->core->getQueries()->getRotatingSectionsForGradeableAndUser($gradeable_id, $this->core->getUser()->getId());
+            $users_to_grade = $this->core->getQueries()->getUsersByRotatingSections($sections);
+            $total = array_sum($this->core->getQueries()->getTotalUserCountByRotatingSections($sections));
+            $graded = array_sum($this->core->getQueries()->getGradedUserCountByRotatingSections($gradeable_id, $sections));
+        }
+
+        if($total == 0) {
+            $total = 0.001;
+        }
+        $progress = round(($graded / $total) * 100, 1);
```

Comment:
I'm not sure what you're trying to do here. If $total is 0 and $graded is some non-zero number (let's say 1), then $progress at a minimum would be `100000` which seems...wrong? I'd clamp this to either 0 or 100 if total is 0 depending on what exactly you're trying to output.